index.ts

```diff
@@ -63,7 +63,15 @@ interface SuiteProto {
 export type SuiteTrait = (this: Mocha.ISuiteCallbackContext, ctx: Mocha.ISuiteCallbackContext, ctor: SuiteCtor) => void;
 export type TestTrait = (this: Mocha.ITestCallbackContext, ctx: Mocha.ITestCallbackContext, instance: SuiteProto, method: Function) => void;
 
-const noname = (cb) => cb;
+const noname = (cb: Function, innerFunction?: Function) => {
```

Comment:
This needs to be a function instead of just const.

Reply:
Not sure I follow?

Reply:
There is an error in the tests that basically boils down to:

```
     TypeError [ERR_INVALID_CALLBACK]: Callback must be a function
```

So I presume that noname does not return the correct result, in fact it might return an empty block or object.

Reply:
I suspect it's to do with the callback being passed in.  If it argument inspection is performed on the function i.e. `noname(...).length ==2? //presume callback`... there might be an issue

Reply:
I have checked out your repository and then augmented the transpiled version of index.ts to also export noname. When calling that it will result in

```
> var i = require('./index');
undefined
> i
{ [Function: tsdd]
  suite: { [Function] skip: [Function], only: [Function], pending: [Function] },
  params:
   { [Function]
     skip: [Function],
     only: [Function],
     pending: [Function],
     naming: [Function] },
  test: { [Function] skip: [Function], only: [Function], pending: [Function] },
  trait: [Function: trait],
  slow: [Function: slow],
  timeout: [Function: timeout],
  retries: [Function: retries],
  skipOnError: { [Function] __mts_isTrait: true },
  pending: [Function: pending],
  only: [Function: only],
  skip: [Function: skip],
  context: [Function: context],
  noname: [Function: noname] }
> i.noname
[Function: noname]
> i.noname()
{ [Function: wrapperResult] toString: [Function] }
> i.noname()()
TypeError: Cannot read property 'apply' of undefined
    at wrapperResult (/Volumes/workspaces/01-coldrye/01-collaboration/tmp/mocha-typescript/index.js:31:18)
    at repl:1:11
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
    at REPLServer.defaultEval (repl.js:239:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:120:20)
    at REPLServer.emit (events.js:210:7)
    at REPLServer.Interface._onLine (readline.js:279:10)
```

As you can see, the return result from noname is not a function.

Reply:
argh, my bad I forgot to pass in the callback. Seems to be working, however it will cause mocha to report an error, still.

Reply:
Ok so `cb` is optional (that's not how I had it defined)

Reply:
No, it is not, I ran the transpiled javascript directly.

Turns out that it is basically boiling to async tests not being run and failing in said TypeError.

Reply:
Ok that sounds more like callback inspection on the arguments

Reply:
@silkentrance this has been corrected, and all tests passed.  TL;DR the wrapped function needed to specify an argument for done if the original function had a done argument otherwise mocha assumed it was a synchronous method